Tools/Cli-LoRa-Device-Provisioning/Tests/LoRaWan.Tools.CLI.Tests.Unit/DeviceProvisioningTest.cs

```diff
@@ -118,18 +142,37 @@ public async Task AddOTAADevice()
                     IsSuccessful = true
                 });
 
-            this.registryManager.Setup(x => x.GetTwinAsync(DevEUI))
-                .ReturnsAsync(savedTwin);
+            // If the device exists in registry we expect the getTwin to return the device.
+            if (deviceExistsInRegistry)
+            {
+                this.registryManager.Setup(x => x.GetTwinAsync(DevEUI)).ReturnsAsync(savedTwin);
+            }
+            else
+            {
+                this.registryManager.SetupSequence(x => x.GetTwinAsync(DevEUI))
+                    .ReturnsAsync((Twin?)null)
+                    .ReturnsAsync(savedTwin);
+            }
 
             // Act
             var args = CreateArgs($"add --type otaa --deveui {DevEUI} --appeui {AppEui} --appkey {AppKey}  --decoder {Decoder} --network {NetworkName}");
             var actual = await Program.Run(args, this.configurationHelper);
 
             // Assert
             Assert.Equal(0, actual);
-            this.registryManager.Verify(c => c.AddDeviceWithTwinAsync(
-                    It.Is<Device>(d => d.Id == DevEUI.ToString()),
-                    It.IsNotNull<Twin>()), Times.Once());
+            if (deviceExistsInRegistry)
+            {
+                this.registryManager.Verify(c => c.UpdateTwinAsync(
```

Comment:
Should we verify that `AddDeviceWithTwinAsync` was never called?
It would apply to L106 too.